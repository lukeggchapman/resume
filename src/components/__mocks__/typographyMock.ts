import idObj from 'identity-obj-proxy'

jest.doMock('../typography', () => ({
  __esModule: true,
  default: idObj,
  FAIcon: 'FAIcon',
}))
