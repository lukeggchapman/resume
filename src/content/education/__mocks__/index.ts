const mockEducationData = [
  {
    frontmatter: {
      school: 'schoolTest1',
      endDate: '2004-05-18',
      degree: 'degreeTest1',
      startDate: '2001-01-11',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
]

export default mockEducationData
