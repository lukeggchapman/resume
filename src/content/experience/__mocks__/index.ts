const mockExperienceData = [
  {
    frontmatter: {
      company: 'companyTest1',
      endDate: 'endDateTest1',
      role: 'roleTest1',
      startDate: 'startDateTest1',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
  {
    frontmatter: {
      company: 'companyTest2',
      endDate: 'endDateTest2',
      role: 'roleTest2',
      startDate: 'startDateTest2',
    },
    logo: 'logoPublcUrlTest2',
    htmlAst: 'htmlAstTest2' as any,
  },
]

export default mockExperienceData
