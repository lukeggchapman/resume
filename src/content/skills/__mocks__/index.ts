export default {
  beginner: ['skillTest1', 'skillTest2'],
  intermediate: ['skillTest3', 'skillTest4'],
  advanced: ['skillTest5', 'skillTest6'],
}
