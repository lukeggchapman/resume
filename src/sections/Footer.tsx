import React from 'react'
import typography, { FAIcon } from 'components/typography'
import { StyleSheet, View, Text, Link as PDFLink } from '@react-pdf/renderer'

const stylesObj = {
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
}
const styles = StyleSheet.create(stylesObj)

export default () => (
  <View style={[typography.h4, styles.footer]}>
    <Text>
      <FAIcon icon="github" fab />
      {` `}
      <PDFLink src="https://github.com/lukeggchapman">lukeggchapman</PDFLink>
    </Text>
    <Text>
      <FAIcon icon="linkedin" fab />
      {` `}
      <PDFLink src="https://www.linkedin.com/in/lukeggchapman">
        lukeggchapman
      </PDFLink>
    </Text>
    <Text>
      <FAIcon icon="react" fab />
      {` `}
      Created with React.
      {' Source code '}
      <PDFLink src="https://bit.ly/2VgE4pH">https://bit.ly/2VgE4pH</PDFLink>
    </Text>
  </View>
)
