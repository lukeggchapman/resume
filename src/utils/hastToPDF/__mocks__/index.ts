export default jest.fn((htmlAst) => htmlAst)
