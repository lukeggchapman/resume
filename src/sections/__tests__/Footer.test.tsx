import React from 'react'
import { shallow } from 'enzyme'
import 'components/__mocks__/typographyMock'

import Footer from '../Footer'

describe('Footer Section', () => {
  const render = () => shallow(<Footer />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
