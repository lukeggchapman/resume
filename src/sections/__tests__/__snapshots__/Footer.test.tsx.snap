// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Section renders correctly 1`] = `
<View
  style={
    Array [
      "h4",
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Text>
    <FAIcon
      fab={true}
      icon="github"
    />
     
    <Link
      src="https://github.com/lukeggchapman"
    >
      lukeggchapman
    </Link>
  </Text>
  <Text>
    <FAIcon
      fab={true}
      icon="linkedin"
    />
     
    <Link
      src="https://www.linkedin.com/in/lukeggchapman"
    >
      lukeggchapman
    </Link>
  </Text>
  <Text>
    <FAIcon
      fab={true}
      icon="react"
    />
     
    Created with React.
     Source code 
    <Link
      src="https://bit.ly/2VgE4pH"
    >
      https://bit.ly/2VgE4pH
    </Link>
  </Text>
</View>
`;
